File: backend/src/main/java/com/joinlivora/backend/report/model/Report.java
package com.joinlivora.backend.report.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "reports", indexes = {
    @Index(name = "idx_report_reporter_id", columnList = "reporter_user_id"),
    @Index(name = "idx_report_reported_id", columnList = "reported_user_id"),
    @Index(name = "idx_report_status", columnList = "status")
})
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Report {

    @Id
    @GeneratedValue
    @UuidGenerator
    @Column(nullable = false, updatable = false)
    private UUID id;

    @Column(name = "reporter_user_id")
    private Long reporterUserId;

    @Column(name = "reported_user_id", nullable = false)
    private Long reportedUserId;

    @Column(name = "stream_id")
    private UUID streamId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportReason reason;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportStatus status;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = Instant.now();
        updatedAt = Instant.now();
        if (status == null) {
            status = ReportStatus.PENDING;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now();
    }
}

File: livora-frontend/src/types/report.ts
export enum ReportReason {
  UNDERAGE = 'UNDERAGE',
  COPYRIGHT = 'COPYRIGHT',
  HARASSMENT = 'HARASSMENT',
  VIOLENCE = 'VIOLENCE',
  NON_CONSENSUAL = 'NON_CONSENSUAL',
  SPAM = 'SPAM',
  OTHER = 'OTHER'
}

export enum ReportStatus {
  PENDING = 'PENDING',
  REVIEWED = 'REVIEWED',
  RESOLVED = 'RESOLVED',
  REJECTED = 'REJECTED'
}

export enum ReportTargetType {
  STREAM = 'STREAM',
  USER = 'USER',
  CHAT_MESSAGE = 'CHAT_MESSAGE'
}

export interface Report {
  id: string;
  reporterUserId?: number;
  reportedUserId: number;
  streamId?: string;
  reason: ReportReason;
  description?: string;
  status: ReportStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReportRequest {
  reportedUserId: number;
  streamId?: string;
  reason: ReportReason;
  description?: string;
}

export interface UpdateReportRequest {
  status: ReportStatus;
}
